include/PathOrderOptimizer.h

```diff
@@ -421,53 +430,65 @@ class PathOrderOptimizer
                 return nullptr;
             };
 
-
-        // We want all outer walls to be printed first (For outside in ordering) or last (for inside out ordering)
-        std::unordered_set<Path> outer_walls = reverse_direction ? leaves : roots;
-
-        if (reverse_direction)
+        if (group_outer_walls)
         {
-            visited.insert(outer_walls.begin(), outer_walls.end());
+            // We want all outer walls to be printed first (For outside in ordering) or last (for inside out ordering)
+            std::unordered_set<Path> outer_walls = reverse_direction ? leaves : roots;
 
-            // Add all inner walls
-            while (!roots.empty())
+            if (reverse_direction)
             {
-                Path root = findClosestPathVertices(current_position, roots);
-                roots.erase(root);
-                actions::dfs(root, order_requirements, handle_node, visited, nullptr, get_neighbours);
-            }
+                visited.insert(outer_walls.begin(), outer_walls.end());
 
-            //Add all outer walls
-            while (!outer_walls.empty())
-            {
-                Path wall = findClosestPathVertices(current_position, outer_walls);
-                outer_walls.erase(wall);
-                handle_node(wall, nullptr);
+                // Add all inner walls
+                while (! roots.empty())
+                {
+                    Path root = findClosestPathVertices(current_position, roots);
+                    roots.erase(root);
+                    actions::dfs(root, order_requirements, handle_node, visited, nullptr, get_neighbours);
+                }
+
+                // Add all outer walls
+                while (! outer_walls.empty())
+                {
+                    Path wall = findClosestPathVertices(current_position, outer_walls);
```

Comment:
I'm seeing this a couple of times in the newly added code and I have some concerns about this. I think we should update the `current_position` once we picked a path. Don't think that is happening right now.

Reply:
Oh no nevermind I see it is updated in the `handle_node` function